main.c

```diff
@@ -240,29 +240,22 @@ void close_info(void)
 
 void open_info(void)
 {
-	int pfd[2];
+	spawn_t pfd;
 	char w[12], h[12];
+	char *argv[5];
 
 	if (info.f.err || info.fd >= 0 || win.bar.h == 0)
 		return;
 	win.bar.l.buf[0] = '\0';
-	if (pipe(pfd) < 0)
-		return;
-	if ((info.pid = fork()) == 0) {
-		close(pfd[0]);
-		dup2(pfd[1], 1);
-		snprintf(w, sizeof(w), "%d", img.w);
-		snprintf(h, sizeof(h), "%d", img.h);
-		execl(info.f.cmd, info.f.cmd, files[fileidx].name, w, h, NULL);
-		error(EXIT_FAILURE, errno, "exec: %s", info.f.cmd);
-	}
-	close(pfd[1]);
-	if (info.pid < 0) {
-		close(pfd[0]);
-	} else {
-		fcntl(pfd[0], F_SETFL, O_NONBLOCK);
-		info.fd = pfd[0];
+	snprintf(w, sizeof(w), "%d", img.w);
+	snprintf(h, sizeof(h), "%d", img.h);
+	construct_argv(argv, ARRLEN(argv), info.f.cmd, files[fileidx].name, w, h, NULL);
```

Comment:
Wouldn't it be cleaner to just have
`char *argv[5] = {info.f.cmd, files[fileidx].name, w, h, NULL}`
than have construct_argv.

Reply:
Don't think that'll be C89 compliant. And if we don't care about that then we can also just go one step further and use compound literals and not declare argv at all.

But given how simple `construct_argv()` is, I'd be leaning more towards keeps things this way.

Reply:
I'll admit I have some bias against va args... dropping c89 support sits well with me. (If we actually care about it though we should amend the build process to actually check).

My c89 isn't the best, but wouldn't we be compliant if we just put all of this in a new scope. Not saying that is better than just dropping support.

But anyways it is just a nit. I'd still approve as things currently stand.